provider/runbook_resource.go

```diff
@@ -121,12 +131,21 @@ func readResourceFireHydrantRunbook(ctx context.Context, d *schema.ResourceData,
 		return diag.Errorf("Error reading runbook %s: %v", runbookID, err)
 	}
 
+	attachmentRule, err := json.Marshal(runbookResponse.AttachmentRule)
+	if err != nil {
+		return diag.Errorf("Error converting step config to JSON due invalid JSON returned by FireHydrant: %v", err)
+	}
+
```

Comment:
could you bring this down to below line 144 to keep all the processing and setting of attributes together?

i also think you might be able to avoid default: null thing by tweaking the logic here a lil bit to be something like this:
```suggestion
	if len(runbookResponse.AttachmentRule) > 0 {
		attachmentRule, err := json.Marshal(runbookResponse.AttachmentRule)
		if err != nil {
			return diag.Errorf("Error converting attachment rule to JSON due invalid JSON returned by FireHydrant: %v", err)
		}
		attributes["attachment_rule"] = string(attachmentRule)
	}
```